### DISCUSSION TITLE
logscale axis support in egui plot?

### DISCUSSION THREAD

**Discussion by hyumo** [emilk/egui#4553] [Q&A] [answered]
Not sure if this is already implemented somewhere? Thanks for your help!

**Comment by YgorSouza** [comment, marked as answer]
There isn't anything out of the box. Issue #2784 is about that but it never got implemented. It might be possible to do that manually by leveraging the [x_grid_spacer](https://docs.rs/egui_plot/0.27.2/egui_plot/struct.Plot.html#method.x_grid_spacer) and [y_grid_spacer](https://docs.rs/egui_plot/0.27.2/egui_plot/struct.Plot.html#method.y_grid_spacer) methods and calling `log10` and `powi(10)` on the data and bounds as needed, but there isn't any example currently. The closest we have is the [Custom Axes](https://github.com/emilk/egui/blob/a8b50e6aa12f7b25cc00c09794a6a5f4dce2fff0/crates/egui_demo_lib/src/demo/plot_demo.rs#L460) demo from the demo app.

Note that there is the [log_grid_spacer](https://docs.rs/egui_plot/0.27.2/egui_plot/fn.log_grid_spacer.html), but all it does is divide the grid recursively as you zoom in (which is already the default behavior). It might serve as a starting point for the log scale grid, though. I'll see if I can give it a go and come up with an example.

**Reply by hyumo** [reply]
Cool thanks, looking forward to the example!

**Reply by YgorSouza** [reply]
I started in #4564, if you want to take a look. It mostly works, but could use some cleanup, and the grid lines don't appear recursively as you zoom in, which is a problem #2784 also ran into. I'll try to see how other plotting libraries handle that and continue polishing the example when I have some more time.

**Reply by hyumo** [reply]
Thanks, I just left a comment in the PR. It looks pretty awesome, except that when x or y is negative, there's no `log10` and showing a lot of `NaN`
